### PR TITLE
Closes #2448 - Use specific CPCSS on frontend Part 6

### LINKED ISSUE #2448: CPCSS for specific pages: Use specific CPCSS on frontend (6/6)
Sub-Card for epic #2377.

This card is for tasks related to the frontend.

#### inc/classes/optimization/CSS/class-critical-css.php
- [x] Update method `get_current_page_critical_css` to use the specific CPCSS for the page if it exists
- [x] Write unit tests
- [x] Write integration tests

### PR DESCRIPTION
**Pull request by crystinutzaa**
Sub-Card for epic #2377.

This card is for tasks related to the frontend.

#### inc/classes/optimization/CSS/class-critical-css.php
- [x] Update method `get_current_page_critical_css` to use the specific CPCSS for the page if it exists
- [x] Write unit tests
- [x] Write integration tests

### CODE REVIEW

File: inc/Engine/CriticalPath/CriticalCSSSubscriber.php [hunk 1]
```diff
@@ -406,7 +406,7 @@ public function insert_load_css() {
 	 * @return string Updated HTML output
 	 */
 	public function insert_critical_css_buffer( $buffer ) {
-		if ( ( defined( 'DONOTROCKETOPTIMIZE' ) && DONOTROCKETOPTIMIZE ) || ( defined( 'DONOTASYNCCSS' ) && DONOTASYNCCSS ) ) {
+		if ( ( rocket_has_constant( 'DONOTROCKETOPTIMIZE' ) && rocket_get_constant( 'DONOTROCKETOPTIMIZE' ) ) || ( rocket_has_constant( 'DONOTASYNCCSS' ) && rocket_get_constant( 'DONOTASYNCCSS' ) ) ) {
```
Comment (Tabrisrp) [inline inc/Engine/CriticalPath/CriticalCSSSubscriber.php:409]:
```suggestion
		if ( rocket_get_constant( 'DONOTROCKETOPTIMIZE' ) || rocket_get_constant( 'DONOTASYNCCSS' ) ) {
```

File: tests/Integration/inc/Engine/CriticalPath/CriticalCSS/getCurrentPageCriticalCss.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace WP_Rocket\Tests\Integration\inc\Engine\CriticalPath\CriticalCSS;
```
Comment (Tabrisrp) [inline tests/Integration/inc/Engine/CriticalPath/CriticalCSS/getCurrentPageCriticalCss.php:3]:
```suggestion
namespace WP_Rocket\Tests\Integration\inc\Engine\CriticalPath\CriticalCSSSubscriber;
```

File: tests/Integration/inc/Engine/CriticalPath/CriticalCSS/getCurrentPageCriticalCss.php [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+use WP_Rocket\Tests\Integration\FilesystemTestCase;
+
+/**
+ * @covers \WP_Rocket\Engine\CriticalPath\CriticalCSS::get_current_page_critical_css
```
Comment (Tabrisrp) [inline tests/Integration/inc/Engine/CriticalPath/CriticalCSS/getCurrentPageCriticalCss.php:8]:
```suggestion
 * @covers \WP_Rocket\Engine\CriticalPath\CriticalCSSSubscriber::insert_critical_css_buffer
```

File: tests/Integration/inc/Engine/CriticalPath/CriticalCSS/getCurrentPageCriticalCss.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+ * @group  CriticalPath
+ * @group  vfs
+ */
+class Test_GetCurrentPageCriticalCSS extends FilesystemTestCase {
```
Comment (Tabrisrp) [inline tests/Integration/inc/Engine/CriticalPath/CriticalCSS/getCurrentPageCriticalCss.php:12]:
```suggestion
class Test_InsertCriticalCssBuffer extends FilesystemTestCase {
```

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by hellofromtonya — APPROVED** [review]

**Review by Screenfeed — APPROVED** [review]

**Comment by Tabrisrp** [comment]
@wp-media/dev-qa ready for QA

**Review by remyperona — APPROVED** [review]

**Comment by arunbasillal** [comment]
- Added `wp-content\cache\critical-css\1\posts\post-1.css`
- The post with the ID `1` used this CPCSS
- Every other post used the default CPCSS

Looking good 👍 

I tried to repeat it for `page` but couldn't make it work. 

@Tabrisrp Is this update only for `posts`? Will there be another one for `pages`? Wondering if I missed something.

**Comment by Tabrisrp** [comment]
@arunbasillal `posts` is the generic folder for any "post type", i.e. it can be a post, a page, any other CPT. The type identified by the filename, using `post_type-id.css`

**Review by arunbasillal — APPROVED** [review]
Thanks @Tabrisrp 

Following this - https://github.com/wp-media/wp-rocket/pull/2541#issuecomment-618431191

- Tested by adding `\wp-content\cache\critical-css\1\posts\page-2.css`
- Page with ID 2 has specific CPCSS
- Other pages had the common CPCSS

All good 👍

**Pull request merged**